File: z4. directory-structure-recursion/directory-recursion.js
var fs = require('fs'),
	l_path = require('path')
	;

var directory_recursion = function(path){
	var file = fs.lstatSync(path),
		json = {
			path: path,
			name: l_path.basename(path)
		}
		;

	if(file.isDirectory()){
		json.type='folder';
		json.children= fs.readdirSync(path).map(function(child){
			return directory_recursion(path+'/'+child);
		});
	} else {
		json.type='file';
	}

	return JSON.stringify(json,null,4);
};

console.log(
	directory_recursion('./root')
);

File: 05. directory-structure-recursion/directory-recursion.py
node = './root'

def directory_recursion(node):
	node = stats_on_node #library?
	json = {
		path: node,
		name: node.name #syntax?
	}

	if(node.isDirectory()):
		json.type = 'folder'
		json.children = file.map(node, lambda child: directory_recursion(node+'/'+child))
	else:
		json.type = 'file'
	print(json)



if __name__ == '__main__':
	directory_recursion(node)
